qualtran/bloqs/chemistry/pbc/first_quantization/prepare_zeta.py

```diff
@@ -0,0 +1,64 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+r"""PREPARE the potential energy terms of the first quantized chemistry Hamiltonian."""
+
+from functools import cached_property
+from typing import Optional, Set, Tuple, TYPE_CHECKING
+
+import numpy as np
+from attrs import frozen
+
+from qualtran import Bloq, Register, Signature
+from qualtran.bloqs.basic_gates import Toffoli
+
+if TYPE_CHECKING:
+    from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+
+
+@frozen
+class PrepareZetaState(Bloq):
+    r"""PREPARE the superpostion over $l$ weighted by $\zeta_l$.
+
+    This is apparently NOT just generic state preparation and there are some
+    tricks I don't understand.
```

Comment:
haha, while I appreciate the candor this might be better as an issue with a link to it

Reply:
Done.